backend/src/main/java/fk/prof/backend/model/assignment/impl/ProcessGroupDetail.java

```diff
@@ -0,0 +1,105 @@
+package fk.prof.backend.model.assignment.impl;
+
+import com.google.common.base.Preconditions;
+import fk.prof.backend.model.assignment.*;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import recording.Recorder;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class ProcessGroupDetail implements ProcessGroupContextForScheduling, ProcessGroupContextForPolling {
+  private static final Logger logger = LoggerFactory.getLogger(ProcessGroupDetail.class);
+
+  private final Recorder.ProcessGroup processGroup;
+  private final int thresholdForDefunctRecorderInSecs;
+  private final Map<RecorderIdentifier, RecorderDetail> recorderLookup = new ConcurrentHashMap<>();
+  private volatile WorkAssignmentSchedule workAssignmentSchedule = null;
+
+  public ProcessGroupDetail(Recorder.ProcessGroup processGroup, int thresholdForDefunctRecorderInSecs) {
+    this.processGroup = Preconditions.checkNotNull(processGroup);
+    this.thresholdForDefunctRecorderInSecs = thresholdForDefunctRecorderInSecs;
+  }
+
+  @Override
+  public Recorder.ProcessGroup getProcessGroup() {
+    return processGroup;
+  }
+
+  /**
+   * Called on http event loop
+   * TODO: Liable for refactoring to implement stickiness behaviour in future
+   * Returns null if poll is refused by recorder detail (if the poll request is obsolete)
+   * Returns null if no work assignment schedule is present
+   * Returns null if recorder cannot accept work because it is defunct or already doing some other work
+   * Returns null if no work assignments are left or none is ready yet to be handed out
+   * Returns null if recorder has been assigned work already
+   * Returns work assignment otherwise
+   * @return
+   */
+  @Override
+  public Recorder.WorkAssignment receivePoll(Recorder.PollReq pollReq) {
+    RecorderIdentifier recorderIdentifier = RecorderIdentifier.from(pollReq.getRecorderInfo());
+    //TODO: clean-up job to remove recorders from lookup which have been defunct for a long long time
+    //The above is only a problem if the same backend stays associated with recorder for a long time, otherwise ProcessGroupDetail will be GC-eligible on de-association
+    RecorderDetail recorderDetail = this.recorderLookup.computeIfAbsent(recorderIdentifier,
```

Comment:
There is a comment above for clean up job. This task has similar requirements as we have for backend detail and removing them past a certain threshold